materialize-postgres-rc/pipeline.go

```diff
@@ -0,0 +1,152 @@
+package main
```

Comment:
The overarching concern is: how do we maximize shared use of (very limited) Postgres connection slots if there are _lots_ of stats materializations going into a single DB?

Even worse, where these stats materializations are running in different datacenters and geographic regions, such that there's non-trivial round-trip latency? (we have this today b/w our managed data-plane in GCP and Supabase in AWS).

We're using pgBouncer, but pgBouncer still needs to hold locks on literal DB connections for the lifetime of a single transaction.

`pgx`, unfortunately, uses a lot of prepared statements (which straight-up don't work with pgBouncer). Even if you turn those off, it's still doing an _awful_ lot of round-tripping to the DB -- even in excess of the literal queries you ask it to run. This means that a single transaction can involve quite a few round-trips.

So, the question is: how do we make transactions require as few round trips as possible? The best possible outcome is that we're able to fire off the entire begin/end block to a pgBouncer session in a single go, so that it can lock a PG connection and apply our transaction batch without _any_ round-trips back to the materialization connector.

That's what this pgBatch implementation achieves. In the typical case we will accumulate all of the transaction statements, including `begin` and `commit`, into a single buffer that's written out to wire all at once. If things get bonkers large then we'll periodically flush out a big batch of statements (which involves holding the transaction connection lock), but this is rare.